pkg/scheduler/framework/preemption/preemption.go

```diff
@@ -401,6 +452,71 @@ func (ev *Evaluator) prepareCandidate(ctx context.Context, c Candidate, pod *v1.
 	return nil
 }
 
+// prepareCandidateAsync triggers a goroutine for some preparation work:
+// - Evict the victim pods
+// - Reject the victim pods if they are in waitingPod map
+// - Clear the low-priority pods' nominatedNodeName status if needed
+// The Pod won't be retried until the goroutine triggered here completes.
+//
+// See http://kep.k8s.io/4832 for how the async preemption works.
+func (ev *Evaluator) prepareCandidateAsync(c Candidate, pod *v1.Pod, pluginName string) *framework.Status {
+	metrics.PreemptionVictims.Observe(float64(len(c.Victims().Pods)))
+
+	// intentionally create a new context, not using a ctx from the scheduling cycle, to create ctx,
+	// because this process could continue even after this scheduling cycle finishes.
+	ctx, cancel := context.WithCancel(context.Background())
+	errCh := parallelize.NewErrorChannel()
+	preemptPod := func(index int) {
+		victim := c.Victims().Pods[index]
+		if err := ev.preemptPod(ctx, c, pod, victim, pluginName); err != nil {
+			errCh.SendErrorWithCancel(err, cancel)
+		}
+	}
+
+	ev.mu.Lock()
+	ev.preempting[pod.UID] = struct{}{}
+	ev.mu.Unlock()
+
+	logger := klog.FromContext(ctx)
+	go func() { // TODO: use paralizer
+		defer cancel()
+
+		// Lower priority pods nominated to run on this node, may no longer fit on
+		// this node. So, we should remove their nomination. Removing their
+		// nomination updates these pods and moves them to the active queue. It
+		// lets scheduler find another place for them.
+		nominatedPods := getLowerPriorityNominatedPods(logger, ev.Handler, pod, c.Name())
+		if err := util.ClearNominatedNodeName(ctx, ev.Handler.ClientSet(), nominatedPods...); err != nil {
```

Comment:
We're not clearing this in `prepareCandidate`(when feature is disabled) ?

Reply:
Currently, we are. 
But, as mentioned in https://github.com/kubernetes/kubernetes/pull/128067, I have to think a bit more about it through the weekend.

Reply:
We should do before.
Because after the moment the high priority pod gets nominated to the node, lower priority pods could be unschedulable, regardless of whether the API calls are completely done or not.

Reply:
But, I still believe we don't need to clear nominated node names of all lower priority pods on the node.
Because, for the extreme example, if high priority pod only requires 2 CPU, and there are 10 lower priority Pods that require 2 CPU each and have already got nominated to the node, it doesn't make sense to remove the nomination from all.

So, I do believe we should seek other way like:
https://github.com/kubernetes/kubernetes/pull/128067

But, I'm considering such enhancement separated from this KEP (i.e., no need to implement it together), and thus I wouldn't like to include the change here.
I was thinking I'd pursue when I've got time after Kubecon. (... or you can take over if you want! :))
Also, either way, I believe that kind of scenario (there's a lower priority pod that just completed the preemption on the same node) is fairly rare, and hence we needn't actively prioritize it.

Reply:
I'm +1 for separate it from this KEP, it will be easily to roll back if sth goes wrong after the implementation.

> ... or you can take over if you want! :

I’ll look into this when I have some free time, but I haven't thought of a good solution yet also :(

Reply:
The rough idea that I have is like:
- When calculating the victims on the node, we assume the nominated Pods are on the node already.
  - Lower priority nominated Pods could be selected as victims, while higher ones shouldn't.
- Then, if lower nominated Pods are selected, `prepareCandidate` (or `prepareCandidateAsync`) unnominates them instead of deleting them.

So, in short, the nominated info on lower pods will be un-nominated only when necessary.

Reply:
But, apart from that idea, I'd like to still seek the simplest option like https://github.com/kubernetes/kubernetes/pull/128067. 
We have to think thoroughly if there's any corner case though, maybe we can just do that.